src/System.Runtime/tests/System/EnumTests.cs

```diff
@@ -732,8 +732,12 @@ public static IEnumerable<object[]> ToObject_InvalidValue_TestData()
         [MemberData(nameof(ToObject_InvalidValue_TestData))]
         public static void ToObject_InvalidValue_ThrowsException(Type enumType, object value, Type exceptionType)
         {
-            ArgumentException ex = (ArgumentException)Assert.Throws(exceptionType, () => Enum.ToObject(enumType, value));
-            Assert.Equal("value", ex.ParamName);
+            if (exceptionType == typeof(ArgumentNullException))
```

Comment:
a cleaner way to do this?
```c#
var type = AssertExtensions.ThrowsAny<ArgumentException>("value", () => Enum.ToObject(enumType, value));
Assert.Equal(exceptionType, type.GetType());
```
and add AssertExtensions.ThrowsAny that's the same as Throws but does Assert.ThrowsAny instead of Assert.Throws inside it.